contracts/src/deployers/rseth-linea/RsETHLineaHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,189 @@
+        // the contribution.
+        else if (_options.asBase) {
+            if (msg.value < _contribution) {
+                revert IHyperdriveDeployerCoordinator.InsufficientValue();
+            }
+            value = _contribution;
+        }
+        // Otherwise, transfer vault shares from the LP and approve the
+        // Hyperdrive pool.
+        else {
+            // NOTE: We don't use `forceApprove` or `safeTransferFrom` since
+            // wrsETH is an OpenZeppelin token contract.
+            //
+            // Take custody of the contribution and approve Hyperdrive to pull
+            // the tokens.
+            IERC20 vaultSharesToken = IERC20(_hyperdrive.vaultSharesToken());
+            bool success = vaultSharesToken.transferFrom(
+                _lp,
+                address(this),
+                _contribution
+            );
+            if (!success) {
+                revert IHyperdriveDeployerCoordinator.TransferFailed();
+            }
+            success = vaultSharesToken.approve(
+                address(_hyperdrive),
+                _contribution
+            );
+            if (!success) {
+                revert IHyperdriveDeployerCoordinator.ApprovalFailed();
+            }
+        }
+
+        return value;
+    }
+
+    /// @notice Convert an amount of vault shares to an amount of base.
+    /// @param _shareAmount The vault shares amount.
+    /// @return The base amount.
+    function convertToBase(uint256 _shareAmount) public view returns (uint256) {
+        return RsETHLineaConversions.convertToBase(rsETHPool, _shareAmount);
+    }
+
+    /// @notice Convert an amount of base to an amount of vault shares.
+    /// @param _baseAmount The base amount.
+    /// @return The vault shares amount.
+    function convertToShares(
+        uint256 _baseAmount
+    ) public view returns (uint256) {
+        return RsETHLineaConversions.convertToShares(rsETHPool, _baseAmount);
+    }
+
+    /// @dev We override the message value check since this integration is
+    ///      payable.
+    function _checkMessageValue() internal view override {}
+
+    /// @notice Checks the pool configuration to ensure that it is valid.
+    /// @param _deployConfig The deploy configuration of the Hyperdrive pool.
+    /// @param _extraData The empty extra data.
+    function _checkPoolConfig(
+        IHyperdrive.PoolDeployConfig memory _deployConfig,
+        bytes memory _extraData
+    ) internal view override {
+        // Perform the default checks.
+        super._checkPoolConfig(_deployConfig, _extraData);
+
+        // Ensure that the vault shares token address is properly configured.
+        if (address(_deployConfig.vaultSharesToken) != rsETHPool.wrsETH()) {
+            revert IHyperdriveDeployerCoordinator.InvalidVaultSharesToken();
+        }
+
+        // Ensure that the base token address is properly configured.
+        if (address(_deployConfig.baseToken) != address(ETH)) {
```

Comment:
do we need this? if we're checking for the eth address shoudln't we also check for the zero address?

Reply:
nvm I can't read